src/injector.js

```diff
@@ -0,0 +1,132 @@
+const puma = require('pumascript');
+const http = require('http');
+const fs = require('fs');
+
+let errorsLogs = [];
+
+/**
+ * Read a .json file with the dependencies to test pumascript.
+ * @param {string} dir - address where the .json file is located.
+ * @return {Object} - object with the content of the json file.
+ */
+function readDependencyJsonFile(dir) {
+    let fileContent;
+    try {
+        fileContent = JSON.parse(fs.readFileSync(dir, 'utf8'));
+    } catch (error) {
+        console.warn(`The file entered does not exist. ${JSON.stringify(error)}`);
+    }
+    return fileContent;
+}
+
+/**
+ * It allows to read the body of a url.
+ * @param {object} options - Object that has to contain host and path 
+ * of the CDN to which you want to access.
+ * @param {fucntion(body)} callback - Function that receives as a parameter the body of the CDN.
+ */
+function readUrl(options, callback) {
+    let req = http.get(options, function (res) {
+        let bodyChunks = [];
+        res.on('data', function (chunk) {
+            bodyChunks.push(chunk);
+        }).on('end', function () {
+            let body = Buffer.concat(bodyChunks);
+            callback(body);
+        })
+    });
+    req.on('error', function (e) {
+        console.error(`ERROR [readURL]:  ${e.message}`);
+    });
+}
+
+/**
+ * Create a file.
+ * @param {string} dir - Address where the file is created.
+ * @param {string} fileName - File name.
+ * @param {string} data - Information that is stored in the file.
+ */
+function createFile(dir, fileName, data) {
+    try {
+        fs.writeFileSync(`${dir}/${fileName}`, data);
+    } catch (error) {
+        console.error(`ERROR [createFile]: ${error.message}`);
+    }
+}
+
+/**
+ *Represents an error encountered when testing a CDN against pumascript.
+ *@constructs 
+ */
+function ErrorTestCDN(url) {
+    this.url = '' || url;    
```

Comment:
The empty string is what is called a _falsey_ value, so, Boolean('') will always evaluate to `false`.

Make `url` to have a default value `url = ''` in case it's undefined. This should suffice.
Else, do `this.url = url || '';` instead or check other conditions within a ternary operation.

Reply:
Ok. I correct it in a moment.